plugins/worldinterface/src/worldproxy.cpp

```diff
@@ -744,6 +744,82 @@ bool WorldProxy::attach(const std::string& id, const std::string& link_name)
     return true;
 }
 
+bool WorldProxy::attachUnscoped(const string& object_name, const std::string& object_link_name, const std::string& robot_name, const string& robot_link_name)
+{
+    gazebo::physics::ModelPtr object_model = world->GetModel(object_name);
+    if(!object_model)
+    {
+        yError() << "GazeboYarpWorldInterface::Attach error --> Object " << object_name << " does not exist in gazebo";
+        return false;
+    }
+    
+    gazebo::physics::LinkPtr object_link = object_model->GetLink(object_link_name);
+    if(!object_link)
+    {
+        yError() << "GazeboYarpWorldInterface::Attach error --> Object link " << object_link_name << " is not found";
+    }
+    
+    gazebo::physics::ModelPtr robot_model = world->GetModel(robot_name);
+    if(!robot_model)
+    {
+        yError() << "GazeboYarpWorldInterface::Attach error --> Robot model " << robot_name << " does not exist in gazebo";
+        return false;
+    }
+    
+    //Get the exact link with only link name instead of full_scoped_link_name
+    gazebo::physics::Link_V robot_model_links = robot_model->GetLinks();
+    for(int i=0; i < robot_model_links.size(); i++)
+    {
+        //E.g iCub::l_hand::l_hand_base_link
+
+        std::string candidate_robot_link_name = robot_model_links[i]->GetScopedName();
+        
+        //Display all the links
+        //yInfo() << "Attach --> Full scoped Candidate robot link name: " << candidate_robot_link_name << " found";
+        
+        std::size_t lastcolon = candidate_robot_link_name.rfind(":");
+        std::string unscoped_robot_link_name = candidate_robot_link_name.substr(lastcolon+1,std::string::npos);
+        //yInfo() << "Attach --> Unscoped robot link name " << unscoped_robot_link_name;
+        
+        if(unscoped_robot_link_name == robot_link_name)
+        {
+            yInfo() << "GazeboYarpWorldInterface::Attach error --> Full scoped Candidate robot link name: " << candidate_robot_link_name << " found";
```

Comment:
This does not seem to be an error, I guess it can be removed.

Reply:
See https://github.com/robotology/gazebo-yarp-plugins/pull/288/commits/2411ce1ba2cddd3415f37af30dc0ce98609567b6